src/Makefile

```diff
@@ -22,24 +22,30 @@ objs-$(CONFIG_SCHEDSTATS) += stats.o
 objs-$(CONFIG_SCHED_DEBUG) += debug.o
 
 obj-m += scheduler.o
+scheduler-objs := $(objs-y) $(sidecar_objs) main.o sched_rebuild.o
+
+search_sym := grep "UND __mod_" | awk '{print substr($$8,7)}' | sort | uniq
+obj-stub := $(addprefix $(obj)/, $(scheduler-objs:.o=.stub.o))
+rdf-file = $(dir $@).$(notdir $@).rdf
 
 CFLAGS_core.stub.o := -DMODULE -DSTACKSIZE_MOD=0
-$(obj)/core.stub.o: $(src)/core.c FORCE
+$(obj)/%.stub.o: $(src)/%.c FORCE
 	$(call cmd,force_checksrc)
 	$(call if_changed_rule,cc_o_c)
 
 GET_STACK_SIZE: $(obj)/core.stub.o
 	$(eval CFLAGS_core.o += -DSTACKSIZE_MOD=$(shell bash $(plugsched_tmpdir)/springboard_search.sh build $<))
 
-$(obj)/.globalize: $(src)/export_jump.h
+$(obj)/.globalize: $(src)/export_jump.h $(obj-stub) FORCE
 	awk -F'[(,]' '{if (/CALLBACK/) {print "__cb_"$$2} else {print $$2}}' $< > $@
+	readelf -sW $(filter %.stub.o, $^) | $(search_sym) >> $@
 
 $(obj)/%.o: $(src)/%.c $(obj)/.globalize GET_STACK_SIZE FORCE
 	$(call cmd,force_checksrc)
 	$(call if_changed_rule,cc_o_c)
-	$(OBJCOPY) --globalize-symbols $(obj)/.globalize $@
+	readelf -sW $@ | grep "UND __mod_" | awk '{print $$8,substr($$8,7)}' > $(rdf-file)
+	$(OBJCOPY) --globalize-symbols $(obj)/.globalize --redefine-syms $(rdf-file) $@
```

Comment:
First, maybe explain a little about `--redefine-syms` in the commit log, for example "this option converts reference to symbol __mod_xxx to xxx (of the module, if there's one)".

Second, there's a potential problem. If we declare `extern int __mod_xxx`, however `xxx` doesn't exist in the module, `__mod_xxx` will refer to `xxx` of the vmlinux eventually.

Finally, @Dengerwei  Please help me check if this patch breaks your GET_STACK_SIZE. I think it doesn't, because all c files are compiled twice, including core.c. But I'm not sure.

Reply:
> Second, there's a potential problem. If we declare extern int __mod_xxx, however xxx doesn't exist in the module, __mod_xxx will refer to xxx of the vmlinux eventually.

This problem will be solved in later.